src/docfx/lib/github/UserProfileCache.cs

```diff
@@ -16,6 +16,7 @@ internal class UserProfileCache
     {
         private readonly ConcurrentDictionary<string, UserProfile> _cacheByName;
         private readonly ConcurrentDictionary<string, UserProfile> _cacheByEmail;
+        private readonly HashSet<string> _unboundEmails = new HashSet<string>();
```

Comment:
HashSet is not thread safe. How about reusing `_cacheByEmail` and use `email` -> `null` to indicate the email does not have a GitHub account

Reply:
As offline discussed, this can be addressed when implementing updating cache.